lib/PersonalityProvider.jsm

```diff
@@ -39,6 +51,43 @@ this.PersonalityProvider = class PersonalityProvider extends UserDomainAffinityP
   get nmfModel() {
     return this.getModel("nmf");
   }
+
+  /**
+   * Returns a Recipe from remote settings to be consumed by a RecipeExecutor.
+   * A Recipe is a set of instructions on how to processes a RecipeExecutor.
+   */
+  getRecipe() {
+    if (!this.recipe) {
+      this.recipe = RemoteSettings("personality-provider-recipe");
+    }
+    return this.recipe;
+  }
+
+  /**
+   * Returns a Text Tagger from a model type, either nb or nmf.
+   * A text tagger allows us to classify text, title or description
+   * of pages found in the browser history.
+   */
+  generateTagger(modelType) {
+    if (modelType !== "nb" && modelType !== "nmf") {
+      throw new Error(`Personality provider received unexpected model for generate tagger: ${modelType}`);
+    }
+
+    if (modelType == "nb") {
+      return new NaiveBayesTextTagger(this.nbModel, new TfIdfVectorizer());
```

Comment:
It's making a bit more sense now, but we don't want like 20 taggers powered by the same model, we want 20 taggers powered by 20 models. Instead of just the model type (nb / nmf) we need to provide the tag as well. So instead saying, "Give me an nb tagger.", we'd say something like "Give me the nb tagger for sports."

Reply:
I see, so how do you imagine that would look in remove settings. 20 models for each nb and nmf? With a total of 40? Or are there some ideas you had around this on the model storage end?